### PR TITLE
Remove IBC-Go versioning table on home readme

### LINKED ISSUE #1041: Update IBC-Go Version table on Readme

### PR DESCRIPTION
**Pull request by boojamya**
Now that we are several releases into Relayer V2, it shouldn't be necessary to maintain the IBC-Go versioning table in the readme anymore. 

This PR also...
-  removes an outdated demo on the home readme and links to the newer one
- fixes make file: `make-gaia`, and `build-gaia`

Closes  https://github.com/cosmos/relayer/issues/1041, https://github.com/cosmos/relayer/issues/871

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,11 +91,17 @@ lint:
...
+CHAIN_CODE := ./chain-code
+GAIA_REPO := $(CHAIN_CODE)/gaia
+
 get-gaia:
-	@mkdir -p ./chain-code/
-	@git clone --branch $(GAIA_VERSION) --depth=1 https://github.com/cosmos/gaia.git ./chain-code/gaia
+	@mkdir -p $(CHAIN_CODE)/
+	@git clone --branch $(GAIA_REPO) --depth=1 https://github.com/cosmos/gaia.git $(GAIA_REPO)
 
 build-gaia:
```
Comment (agouin) [inline Makefile:104]: since we removed the legacy tests in favor of the ibctest cases, are these relevant to the repo anymore?

Comment (boojamya) [reply Makefile:104]: To my knowledge they are only used if people try and run the `dev-env` script in `examples/demo`

If the gaiad binary doesn't exists, it will recommend you run 'make build-gaia'

Comment (boojamya) [reply Makefile:104]: Could also remove these make commands and add a small blurb in the demo readme on how to download and build  Gaia.

**Review by agouin — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (agouin) [reply Makefile:104]: maybe we should replace the demo examples with ibctest examples? Seems fine to keep this for now though. Thanks for fixing!

Comment (boojamya) [reply Makefile:104]: Yes, however the current demos are such an easy example showing the basics of relaying and IBC. `ibctest` adds a another level complexity. 
For that reason, I'm in favor of keeping the demos in.
